internal/services/report_processor.go

```diff
@@ -117,44 +117,48 @@ func ProcessReport(msg *kafka.Message) {
 				continue
 			}
 
-			usage_data_byte, err := kruize.Update_results(experiment_name, k8s_object)
-			if err != nil {
-				log.Error(err)
-				continue
-			}
-
-			for _, data := range usage_data_byte {
+			k8s_object_chunks := utils.SliceK8sObjectToChunks(k8s_object)
```

Comment:
IMO it would be logically straight-forward to compare `len(k8s_objects)` and `cfg.KruizeMaxBulkChunkSize` before using this util function, such that we use it only if we need to.

Reply:
No we cannot because `k8s_object` is of type `[]map[string]interface{}` and  `k8s_object_chunks` is of type `[][]map[string]interface{}`.

Reply:
If we use the util function only if len(k8s_objects) > 100 then I think we could set k8s_object as [][]map[string]interface{} on another var for easy looping

Reply:
Done, Made the changes.